adaboost/cuda/core/operations_impl.cu

```diff
@@ -44,37 +44,49 @@ namespace adaboost
                     (vec.get_data_pointer(gpu), vec.get_size(gpu), value);
                 }
             }
-
+    
             template <class data_type_matrix>
-            __global__ void fill_matrix_kernel
-            (data_type_matrix* data, unsigned cols, data_type_matrix value)
-            {
-                unsigned row = blockDim.y*blockIdx.y + threadIdx.y;
-                unsigned col = blockDim.x*blockIdx.x + threadIdx.x;
-                data[row*cols + col] = value;
+            __global__
+            void thread_multi(data_type_matrix *t1, data_type_matrix value, unsigned total_elements){
+                unsigned i = blockDim.x * blockIdx.x + threadIdx.x;
+                if(i < total_elements)
+                    t1[i] = value;
             }
 
             template <class data_type_matrix>
-            void fill(const data_type_matrix value, const MatrixGPU<data_type_matrix>& mat, unsigned block_size_x, unsigned block_size_y)
+            void fill(const data_type_matrix value, const MatrixGPU<data_type_matrix>& mat, unsigned num_streams)
             {
-                bool gpu=true;
-                if(block_size_x == 0 || block_size_y == 0)
-                {
-                    adaboost::core::fill(value, mat);
+                bool gpu = true;
+                unsigned N=mat.get_cols();
+                unsigned total_elements=mat.get_cols()*mat.get_rows();
+                
+                //array of stream objects
+                cudaStream_t stream[num_streams];
+
+                for(int i = 0; i < num_streams; i++) {
+                    cudaStreamCreate(&stream[i]);
```

Comment:
Create wrappers for these CUDA APIs in cuda_utils and then use those wrappers here.

Reply:
Done